airbyte-lib/airbyte_lib/telemetry.py

```diff
@@ -0,0 +1,63 @@
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+
+import datetime
+import os
+from dataclasses import asdict, dataclass
+from enum import Enum
+from typing import Any, Dict, Optional
+
+import requests
+from airbyte_lib.version import get_version
+
+# TODO: Use production tracking key
+TRACKING_KEY = "jxT1qP9WEKwR3vtKMwP9qKhfQEGFtIM1" or str(os.environ.get("AIRBYTE_TRACKING_KEY"))
+
+
+class SourceType(str, Enum):
+    VENV = "venv"
+    LOCAL_INSTALL = "local_install"
+
+
+class CacheType(str, Enum):
+    # TODO: Extend once proper caches have been merged
+    IN_MEMORY = "in_memory"
+    STREAMING = "streaming"
```

Comment:
Even though it wouldn't contain much, we could also adopt the "TelemetryInfo" convention to be parallel with the source side. In that case, this would be a "CacheTelemetryInfo" dataclass, perhaps just with one member, and would render as a dict instead of a string in the telemetry object.

Reply:
> Since in-memory is out of scope for now, what do you think of the cache types just being the name of the DB backed? So, something like DUCKDB, POSTGRES, SNOWFLAKE, etc

Yes, that's what the TODO on top is meant for.

> Even though it wouldn't contain much, we could also adopt the "TelemetryInfo" convention to be parallel with the source side. In that case, this would be a "CacheTelemetryInfo" dataclass, perhaps just with one member, and would render as a dict instead of a string in the telemetry object.

Good suggestion, I will adjust it like that - maybe we want to add other dimensions later on.